ping/ping.c

```diff
@@ -1773,9 +1795,13 @@ int ping4_parse_reply(struct ping_rts *rts, struct socket_st *sock,
 	if (!rts->opt_flood) {
 		pr_options(rts, opts, olen + sizeof(struct iphdr));
 
-		putchar('\n');
-		fflush(stdout);
+		// handled by print_json()
+		if(!rts->opt_json) {
+			putchar('\n');
+			fflush(stdout);
+		}
 	}
```

Comment:
Generally I would rather vote for adding an API for the generic output and do all the stuff there.

In this place we should call a function `output_finish_line()` or something like that and keep the code that deals with `stdout` in the `ping_output.h` or `ping_output.c`.

And the same with the version, we should just have a function called `output_ping_version()` call it from here and let the code in `ping_output.c` do the work. The code in `ping.c` is way to complex already adding more code there will not help.

Reply:
Thanks, good idea, will adapt.